### PR TITLE
Homework 2 tests

### PR DESCRIPTION
**Pull request by MlPablo**
1. Added unit tests for services, repo, http handlers. All tests in the same pack as there logic.
2. Added integration and e2e tests. They are located at ./tests dir.

### CODE REVIEW

File: tests/e2e/email_service_test.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package e2e
+
+// func (suite *E2ESuite) TestSendEmail() {
```
Comment (vorobeiDev) [inline tests/e2e/email_service_test.go:3]: Is it no a needed file?

**Review by vorobeiDev — APPROVED** [review]

File: tests/e2e/email_service_test.go (hunk 1 above)
Comment (MlPablo) [reply tests/e2e/email_service_test.go:3]: It's not completed tests. In the next commit will fix this

Comment (MlPablo) [reply tests/e2e/email_service_test.go:3]: Fixed in 3-hw pr.

File: tests/e2e/suite.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
+package e2e
```
Comment (oleksandr-kh) [inline tests/e2e/suite.go:1]: чому в e2e не зробив тест на /rate ? це було б найкраще рішення саме для тестування реальної середи

File: tests/functional/service_tests.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+import (
+	"context"
+
+	"github.com/bxcodec/faker/v3"
+
+	"exchange/internal/domain"
+)
+
+func (suite *Suite) TestValidCreateUser() {
```
Comment (oleksandr-kh) [inline tests/functional/service_tests.go:11]: так виглядає, що тут дублюється перевірки з e2e
я не думаю що в цьому є великий сенс

**Review by oleksandr-kh — CHANGES_REQUESTED** [review]
загалом робота досить непогано, добре попрацював 👍

File: internal/repository/filesystem/file_test.go [hunk 4]
```diff
@@ -36,33 +37,12 @@ func TestFileSave(t *testing.T) {
...
-	require.NoError(t, err)
-
-	defer os.Remove(testFilePath)
-
-	testEmail := faker.Email()
-
-	err = repo.SaveUser(ctx, domain.NewUser(testEmail))
-	require.NoError(t, err)
-
-	email, err := repo.GetByEmail(ctx, testEmail)
```
[the remark this answers is not in the document]
Comment (MlPablo) [reply internal/repository/filesystem/file_test.go:52]: Fixed this in third home work.

File: tests/e2e/suite.go (hunk 2 above)
Comment (MlPablo) [reply tests/e2e/suite.go:1]: Made this in 3 home work.

File: tests/functional/service_tests.go (hunk 3 above)
Comment (MlPablo) [reply tests/functional/service_tests.go:11]: Yes, you are right. I thought that functional test must have near the same logic, but with difference in response. Ok, i'll delete this one.

**Comment by MlPablo** [comment]
The commit with the resolve is in 3 homework pr. This was done due to some refactors, so it's better to push changes into new branch. For now I merge this branch.

**Pull request merged**